app/api/auth/oauth.py

```diff
@@ -0,0 +1,78 @@
+# app/auth/oauth.py
+
+from uuid import UUID
+from dotenv import load_dotenv
+from fastapi import Depends, HTTPException, status
+from fastapi.security import OAuth2PasswordBearer
+import os
+import jwt
+from app.models import TokenData
+from app.models import User
+from app.api.auth.auth import fake_db_user
+
+load_dotenv()
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/api/v1/login/")
+
+async def get_current_user(token: str = Depends(oauth2_scheme)):
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Could not validate credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    try:
+        payload = jwt.decode(token, os.getenv("SECRET_KEY"), algorithms=[os.getenv("ALGORITHM")])
+        user_id: str = payload.get("sub")
```

Comment:
Instead of using the sub like that, use TokenData to validate it. Make sure the received data schema is equivalent. Also, check if it's a valid UUID.

Reply:
if you used TokenData, it will expect to have user_id in the Token payload. So, you should use user_id instead of sub. Or you could update user_id in TokenData and call it sub, up to you.

Reply:
Don't forget to add an expiration date and validation to it